Ormael7/Corruption-of-Champions
# Combat Ability API

This is a medium-size refactoring that organizes abilities into their own hierarchy of classes. 

Every ability should be put in its own class where it defines:
* how to test if player knows the ability
* how to test if player can use the ability right now against the monster
* resource costs
* damage/effect calculation and application

See documentation in CombatAbility class and examples. A lot of common checks and adjustments have been moved to AbstractSpell/AbstractWhiteSpell classes.

In this versions, only White spells are converted. However, new spells of other schools could be made with new system already.

Additionally:
- fixed int overflow in big damage display.
- in debug mode, added "Cheat Ability" menu to invoke any new-system ability bypassing knowledge and usability checks
- added Abilities menu to Stats panel, listing all known abilities.
- scalingBonusXXX takes optional randomize argument, default true. If false, will return fixed number, using average random bonus
